owltorrent-network/src/main/java/edu/rice/owltorrent/network/messagereader/BusyWaitMessageReader.java

```diff
@@ -0,0 +1,60 @@
+package edu.rice.owltorrent.network.messagereader;
+
+import edu.rice.owltorrent.network.MessageReader;
+import edu.rice.owltorrent.network.PeerMessage;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.util.Arrays;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Blocking implementation of message reader.
+ *
+ * @author Max Yu
+ */
+@Log4j2(topic = "general")
+public class BusyWaitMessageReader implements MessageReader {
+  private static final int DEF_BUFFER_SIZE = 16;
+
+  // 2 MB
+  private static final int MAX_MESSAGE_SIZE = 2 * 1024 * 1024;
+
+  @Override
+  public PeerMessage readMessage(ReadableByteChannel inputChannel) throws IOException {
+    ByteBuffer buffer = ByteBuffer.allocate(DEF_BUFFER_SIZE);
+    buffer.limit(PeerMessage.LENGTH_FIELD_SIZE);
+    int readBytes = inputChannel.read(buffer);
+    if (readBytes < 0) {
+      log.debug("connection is closed by other peer");
+      return null;
+    }
+    int pstrLength = buffer.getInt(0);
```

Comment:
What does this stand for? Consider renaming

Reply:
It stands for peer string